sublime_jedi/daemon.py

```diff
@@ -17,21 +18,24 @@
 DAEMONS = defaultdict(dict)  # per window
 
 
-def _prepare_request_data(view, location):
-    if location is None:
-        location = view.sel()[0].begin()
-    current_line, current_column = view.rowcol(location)
-
-    filename = view.file_name() or ''
-    source = view.substr(sublime.Region(0, view.size()))
-    return filename, source, current_line, current_column
+def _get_daemon(view):
+    project_path = _find_project(view)
+    if project_path not in DAEMONS:
+        DAEMONS[project_path] = Daemon(
+            project_path=project_path,
+            settings=get_settings(view)
+        )
+    return DAEMONS[project_path]
 
 
-def _get_daemon(view):
-    window_id = view.window().id()
-    if window_id not in DAEMONS:
-        DAEMONS[window_id] = Daemon(settings=get_settings(view))
-    return DAEMONS[window_id]
+def _find_project(view):
+    directory = up(abspath(view.file_name()))
+    while '__init__.py' in os.listdir(directory) and directory != '/':
+        directory = up(directory)
+    if directory == '/':
+        return up(abspath(view.file_name()))
+    else:
+        return directory
```

Comment:
what in case when file just created, but not saved  and it does not have any name?

Reply:
@edelvalle your thoughts?

Reply:
True, let me fix this.

Reply:
It seems this was not fixed in the end:
```python
  File "/home/sylvain/.config/sublime-text-3/Packages/Jedi - Python autocompletion/sublime_jedi/daemon.py", line 32, in _find_project
    directory = up(abspath(view.file_name()))
  File "./python3.3/posixpath.py", line 381, in abspath
  File "./python3.3/posixpath.py", line 64, in isabs
AttributeError: 'NoneType' object has no attribute 'startswith'
```